src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx

```diff
@@ -253,18 +253,37 @@ export default class GeneralUserSettingsTab extends React.Component<IProps, ISta
         PlatformPeg.get()?.setSpellCheckEnabled(spellCheckEnabled);
     };
 
-    private onPasswordChangeError = (err: { error: string } & MatrixError): void => {
-        // TODO: Figure out a design that doesn't involve replacing the current dialog
-        let errMsg = err.error || err.message || "";
-        if (err.httpStatus === 403) {
-            errMsg = _t("Failed to change password. Is your password correct?");
-        } else if (!errMsg) {
-            errMsg += ` (HTTP status ${err.httpStatus})`;
+    private onPasswordChangeError = (err: Error): void => {
+        logger.error("Failed to change password: " + err);
+
+        let underlyingError = err;
+        if (err instanceof UserFriendlyError && err.cause instanceof Error) {
+            underlyingError = err.cause;
         }
-        logger.error("Failed to change password: " + errMsg);
+
+        const errorMessage = extractErrorMessageFromError(
+            err,
+            _t("Unknown password change error (%(stringifiedError)s)", {
+                stringifiedError: String(err),
+            }),
+        );
+
+        let errorMessageToDisplay;
+        if (underlyingError instanceof HTTPError && underlyingError.httpStatus === 403) {
+            errorMessageToDisplay = _t("Failed to change password. Is your password correct?");
+        } else if (underlyingError instanceof HTTPError) {
+            errorMessageToDisplay = _t("%(errorMessage)s (HTTP status %(httpStatus)s)", {
+                errorMessage,
+                httpStatus: underlyingError.httpStatus,
+            });
+        } else {
+            errorMessageToDisplay = errorMessage;
+        }
```

Comment:
What would you think about doing something along the lines of:
```
let errorMessageToDisplay = errorMessage;
if(it is an HTTPError) {
  if(it is a 403) {
    reassign errorMessageToDisplay
  } else {
    reassign errorMessageToDisplay
  } 
}
```
I'm not completely sold on whether the nested if is better than what you have, but I'm a fan of being able to get rid of the last else with the initial assignment of `errorMessageToDisplay` at least.

Reply:
Both are 🤷 type of problems to me in this case.

A bigger better change would be to remove the `else if (underlyingError instanceof HTTPError)` case altogether in favor of [translated `HTTPError` in the `matrix-js-sdk`](https://github.com/matrix-org/matrix-js-sdk/issues/1309) as we're not providing much value by just translating `... (HTTP status ...)` here.

> but I'm a fan of being able to get rid of the last else with the initial assignment of `errorMessageToDisplay` at least.

I've updated to do this ⏩